flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -230,8 +242,15 @@ public void open() {
             if (line == null) {
                 continue;
             }
+
             final Optional<Operation> operation = parseCommand(line);
-            operation.ifPresent(this::callOperation);
+            if (operation.isPresent()) {
+                boolean success = callOperation(operation.get());
```

Comment:
How about letting all the `callXXXX` methods throws `SqlExecutionException` instead of returning flag?
I noticed the error handling logic in every `callXXXX` is the same:
```
        catch (SqlExecutionException e) {
            printExecutionException(e);
            return false;
        }
```
Thus I think it would be also enough to handle the exception here, and you can check whether it's interactive mode and do things you want.